### PR TITLE
2048: Animations

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
Tiles now slide around merrily and new ones pop in. It's a delight.

https://user-images.githubusercontent.com/18616863/121723684-7f7fab80-caef-11eb-9a0b-e91844790f82.mp4

Drive-by changes:
* Convert 2048 to east const
* `LibGfx::Rect::centered_on(Point, Size)`

### CODE REVIEW

File: Userland/Games/2048/Game.h [hunk 1]
```diff
@@ -44,7 +45,10 @@ class Game final {
         void add_tile(size_t row, size_t column, u32 value)
         {
             tiles[row][column] = value;
+            last_added_position = Tuple { row, column };
         }
+
+        Tuple<size_t, size_t> last_added_position { 0, 0 };
```
Comment (alimpfard) [inline Userland/Games/2048/Game.h]: Could we please not use `Tuple` in normal code?
it tends to make life/understanding very difficult in normal cases

Comment (alimpfard) [reply Userland/Games/2048/Game.h]: I originally added `Tuple` purely for metaprogramming tasks, and I think it should stay that way

Comment (dmitrii-ubskii) [reply Userland/Games/2048/Game.h]: I will say that I originally used Tuple because I expected it to already have `operator==()` implemented and for no other reason. But yes, fair point.

Comment (alimpfard) [reply Userland/Games/2048/Game.h]: Thanks for keeping the code readable lol

File: Userland/Libraries/LibGfx/Rect.h [hunk 2]
```diff
@@ -422,6 +422,11 @@ class Rect {
 
     void intersect(const Rect<T>&);
 
+    static Rect<T> centered(const Point<T>& center, const Size<T>& size)
```
Comment (alimpfard) [inline Userland/Libraries/LibGfx/Rect.h:425]: `centered_on()`? I think there's a `center_within()` method, so the generic sounding name might be confusing

File: Userland/Games/2048/Game.h [hunk 3]
```diff
@@ -41,10 +41,23 @@ class Game final {
 
         Tiles tiles;
 
+        struct Position {
+            size_t row;
+            size_t column;
+
+            bool operator==(Position const& other)
```
Comment (alimpfard) [inline Userland/Games/2048/Game.h:51]: This method could be made `const`.

File: Userland/Games/2048/BoardView.cpp [hunk 4]
```diff
@@ -24,6 +24,7 @@ void BoardView::set_board(Game::Board const* board)
     if (has_timer())
         stop_timer();
```
Comment (alimpfard) [inline Userland/Games/2048/BoardView.cpp:26]: (unrelated to code)
Please make the commit message an actual sentence, e.g. "2048: Add sliding animation"

**Review by alimpfard — CHANGES_REQUESTED** [review]
Very cool, just a few small nitpicks

File: Userland/Games/2048/Game.h (hunk 3 above)
Comment (dmitrii-ubskii) [reply Userland/Games/2048/Game.h:51]: Good catch!

**Review by alimpfard — APPROVED** [review]
LGTM

**Comment by awesomekling** [comment]
Oh wow, `Rect::centered_on()` is great! :)
